src/server.rs

```diff
@@ -165,7 +168,17 @@ impl UdpServer {
 
     /// Gossip to one other UDP server.
     async fn gossip(&self, addr: impl Into<String>) -> anyhow::Result<()> {
-        let syn = self.scuttlebutt.lock().await.create_syn_message();
+        let mut scuttlebutt = self.scuttlebutt.lock().await;
+        let heartbeat = scuttlebutt
+            .self_node_state()
+            .get(HEARTBEAT_KEY)
+            .and_then(|heartbeat| str::parse(heartbeat).ok())
+            .unwrap_or(0);
+        scuttlebutt
+            .self_node_state()
+            .set(HEARTBEAT_KEY, heartbeat + 1);
```

Comment:
I'd say parsing a string here is pretty ugly, but I'd assume the functionality is mostly what you were thinking of?

Reply:
Yeah let's not parse it.
What about having a local heartbeat in the object and auto increment it?
We can make it an atomic if it helps.